parlai/core/metrics.py

```diff
@@ -444,7 +456,7 @@ def _drain_queue(self):
             return
         while not self._queue.empty():
             try:
-                yield self._queue.get()
+                yield self._queue.get(timeout=0.05)
```

Comment:
Not familiar with this queue. Is this timeout to avoid race conditions or something like that?